sunbeam-python/sunbeam/core/watcher.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import time
+from typing import Any
+
+import timeout_decorator
+from watcherclient import v1 as watcher
+from watcherclient.common.apiclient.exceptions import NotFound
+from watcherclient.v1 import client as watcher_client
+
+from sunbeam.core.common import SunbeamException, read_config
+from sunbeam.core.deployment import Deployment
+from sunbeam.core.juju import JujuHelper
+from sunbeam.core.openstack_api import get_admin_connection
+from sunbeam.steps.openstack import REGION_CONFIG_KEY
+
+LOG = logging.getLogger(__name__)
+
+# Timeout while waiting for the watcher resource to reach the target state.
+TIMEOUT = 60 * 3
+# Sleep interval between querying watcher resources.
+SLEEP_INTERVAL = 5
```

Comment:
It could be useful to include the units in comments here. eg. seconds, minutes, milliseconds?